### PR TITLE
feat: [SRM-15180]: completing the phase 2 of Azure metrics implementation.

### PR DESCRIPTION
**Pull request by harshilbits**
### Summary
1. Azure metric complete implementation.
2. All changes around templates support for Azure logs and Azure metrics.

#### Screenshots
New Note
New Note

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by harshilbits** [comment]
retrigger titlecheck

**Comment by praneshg239** [comment]
image

@harshilbits Do we need all these fields as full width?

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/components/CommonQueryViewer/components/FieldsToFetchRecords/__tests__/FieldsToFetchRecords.test.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import { FIELD_ENUM } from '@cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSource.constants'
+import {
+  CommonCustomMetricFormikInterface,
+  FieldMapping
+} from '@cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSource.types'
+import { getIsAnyFieldNonFixed, getIsFetchRecordsHidden } from '../FieldsToFetchRecords.utils'
+
+// Test getIsAnyFieldNonFixed function
+test('getIsAnyFieldNonFixed should return true when there is at least one non-fixed field', () => {
+  const fieldsToFetchRecords: FieldMapping[] = [
```
Comment (deepesh-ui) [inline src/modules/85-cv/components/CommonQueryViewer/components/FieldsToFetchRecords/__tests__/FieldsToFetchRecords.test.tsx:10]: move fieldsToFetchRecords to mock file, its being defined in multiple tests

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationInputSetStep/components/TemplatisedRunTimeMonitoredService/TemplatisedRunTimeMonitoredService.tsx [hunk 2]
```diff
@@ -95,12 +96,12 @@ export default function TemplatisedRunTimeMonitoredService(
                         name={`${prefix}spec.monitoredService.spec.templateInputs.${input.path}`}
                         label={getString('connector')}
                         placeholder={getString('cv.healthSource.connectors.selectConnector', {
-                          sourceType: getSourceTypeForConnector(healthSource)
+                          sourceType: sourceType
```
Comment (deepesh-ui) [inline src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationInputSetStep/components/TemplatisedRunTimeMonitoredService/TemplatisedRunTimeMonitoredService.tsx:99]: { sourceType }
```suggestion
                          sourceType
```

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSource.utils.ts [hunk 3]
```diff
@@ -619,7 +662,10 @@ export function getFieldName(
   getString: UseStringsReturn['getString'],
   name?: string
 ): string {
-  if (name === HealthSourceTypes.AzureLogs && fieldIdentifier === CustomMetricFormFieldNames.INDEX) {
+  if (
+    (name === HealthSourceTypes.AzureLogs || name === HealthSourceTypes.AzureMetrics) &&
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSource.utils.ts:666]: move this logic inside `case CustomMetricFormFieldNames.INDEX:`

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.utils.ts [hunk 4]
```diff
@@ -66,8 +85,8 @@ export function generateHealthSourceQueryParams({
   fieldsToFetchRecords,
   values
 }: {
-  identifier: string
-  queryFieldValue: string | undefined
+  identifier?: string
+  queryFieldValue?: string | undefined
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/components/CustomMetricForm/components/CommonCustomMetricFormContainer/CommonCustomMetricFormContainerLayout/CommonCustomMetricFormContainer.utils.ts:89]: queryFieldValue?: string

File: src/modules/85-cv/components/CommonQueryViewer/components/FieldsToFetchRecords/FieldsToFetchRecords.tsx [hunk 5]
```diff
@@ -39,15 +50,17 @@ export default function FieldsToFetchRecords(props: FieldsToFetchRecordsProps):
...
-        <Button
-          variation={ButtonVariation.SECONDARY}
-          text={getString('cv.monitoringSources.gcoLogs.fetchRecords')}
-          onClick={handleFetchRecords}
-          disabled={isQueryButtonDisabled}
-          tooltip={runQueryBtnTooltip}
-        />
-      </Layout.Horizontal>
+      {hideFetchRecordsButton ? null : (
+        <Layout.Horizontal spacing={'large'}>
```
[the remark this answers is not in the document]
Comment (harshilbits) [reply src/modules/85-cv/components/CommonQueryViewer/components/FieldsToFetchRecords/FieldsToFetchRecords.tsx:54]: done.

File: src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationInputSetStep/components/TemplatisedRunTimeMonitoredService/TemplatisedRunTimeMonitoredService.tsx (hunk 2 above)
Comment (harshilbits) [reply src/modules/85-cv/components/PipelineSteps/ContinousVerification/components/ContinousVerificationInputSetStep/components/TemplatisedRunTimeMonitoredService/TemplatisedRunTimeMonitoredService.tsx:99]: done.

File: src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSource.utils.ts (hunk 3 above)
Comment (harshilbits) [reply src/modules/85-cv/pages/health-source/connectors/CommonHealthSource/CommonHealthSource.utils.ts:666]: done.

**Comment by harshilbits** [comment]
> image
> @harshilbits Do we need all these fields as full width?

@praneshg239  , other fields what we show are also full width i.e queryField etc.

**Review by praneshg239 — APPROVED** [review]

**Review by deepesh-ui — APPROVED** [review]

**Pull request merged**